cupy/linalg/_eigenvalue.py

```diff
@@ -35,30 +39,57 @@ def _syevd(a, UPLO, with_eigen_vector):
     else:  # UPLO == 'U'
         uplo = cublas.CUBLAS_FILL_MODE_UPPER
 
-    if dtype == 'f':
-        buffer_size = cupy.cuda.cusolver.ssyevd_bufferSize
-        syevd = cupy.cuda.cusolver.ssyevd
-    elif dtype == 'd':
-        buffer_size = cupy.cuda.cusolver.dsyevd_bufferSize
-        syevd = cupy.cuda.cusolver.dsyevd
-    elif dtype == 'F':
-        buffer_size = cupy.cuda.cusolver.cheevd_bufferSize
-        syevd = cupy.cuda.cusolver.cheevd
-    elif dtype == 'D':
-        buffer_size = cupy.cuda.cusolver.zheevd_bufferSize
-        syevd = cupy.cuda.cusolver.zheevd
+    global _cuda_runtime_version
+    if _cuda_runtime_version < 0:
+        _cuda_runtime_version = runtime.runtimeGetVersion()
+
+    if (not runtime.is_hip and _cuda_runtime_version >= 11020):
+        if dtype.char not in 'fdFD':
+            raise RuntimeError('Only float and double and cuComplex and '
+                               + 'cuDoubleComplex are supported')
```

Comment:
I see that it's a legacy mistake, could you change to use NumPy types instead of C types? 
```suggestion
            raise RuntimeError('Only float32, float64, complex64, and '
                               'complex128 are supported')
```